src/Cli/OrgCommands.php

```diff
@@ -132,29 +122,127 @@ public function orgAnalyze($org, $options = ['as' => 'default', 'format' => 'tab
+            $data = $api->gitHubAPI()->api('repo')->contents()->show($org, $repoName, 'CODEOWNERS');
+            if (!empty($data['content'])) {
+                $content = base64_decode($data['content']);
+                $ownerSource = 'file';
+                $codeowners = static::filterGlobalCodeOwners($content);
+            }
+        } catch (\Exception $e) {
+        }
+
+        return static::inferOwners($api, $org, $repoName, $codeowners, $ownerSource);
+    }
+
+    /**
+     * @command org:update-projects-info
+     * @param $csv_file The path to csv file that contains projects information.
+     */
+    public function orgUpdateProjectsInfo($csv_file, $options = [
+        'as' => 'default',
+        'update-codeowners' => false,
+        'update-support-level-badge' => false,
+        'branch-name' => 'project-update-info',
+        'commit-message' => 'Update project information.',
+        'pr-body' => '',
+    ])
+    {
+        $api = $this->api($options['as']);
+        $updateCodeowners = $options['update-codeowners'];
+        $updateSupportLevelBadge = $options['update-support-level-badge'];
+        if (!$updateCodeowners && !$updateSupportLevelBadge) {
+            throw new \Exception("Either --update-codeowners or --update-support-level-badge must be specified.");
+        }
+        $prTitle = '[UpdateTool - Project Information] Update project information.';
+        $branchName = $options['branch-name'];
+        $commitMessage = $options['commit-message'];
+        if (file_exists($csv_file)) {
+            $csv = new \SplFileObject($csv_file);
+            $csv->setFlags(\SplFileObject::READ_CSV);
+            foreach ($csv as $row_id => $row) {
+                // Skip header row.
+                if ($row_id == 0) {
+                    continue;
+                }
+                $projectUpdateSupportLevel = $updateSupportLevelBadge;
+                $projectUpdateCodeowners = $updateCodeowners;
+                $projectFullName = $row[3];
+                $projectOrg = $row[5];
+                $projectSupportLevel = $row[23];
+                $projectDefaultBranch = $row[97];
+                $codeowners = '';
+                $ownerSource = '';
+                if ($this->validateProjectFullName($projectFullName) && !empty($projectDefaultBranch) && !empty($projectOrg)) {
+                    // If empty or invalid support level, we won't update it here.
+                    if ($projectUpdateSupportLevel && (empty($projectSupportLevel) || !$this->validateProjectSupportLevel($projectSupportLevel))) {
+                        $projectUpdateSupportLevel = false;
+                    }
+                    if ($projectUpdateCodeowners) {
+                        list($codeowners, $ownerSource) = $this->guessCodeowners($api, $projectOrg, $projectFullName);
+                        if (empty($codeowners)) {
+                            // @todo: Should we decide a course of action based on $ownerSource?
```

Comment:
$ownerSource is just something like `file`, it won't be set if $codeowners is empty

Reply:
It looks like it could be "file", "api" or "guess". Then, my question is, should it be the same result no matter the source?

Reply:
Probably for now. Having a guess / don't guess option might be handy. Optional.

Reply:
If it's "file" then it's already there and updating it doesn't make sense. So, I guess we're down to updating it only if "api" or "guess". Maybe adding --only-api and --only-guess options?

Reply:
Implemented new codeowners-only-api and codeowners-only-guess options.